### PR TITLE
use weldx-widgets k3d implementation if available

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
- optionally use code from `weldx-widgets` if installed

## Related Issues

Closes # (add issue numbers)

## Checks

- [ ] updated CHANGELOG.rst
- [ ] updated tests
- [ ] updated doc/
- [ ] update example/tutorial notebooks
- [ ] update manifest file

### CODE REVIEW

File: weldx/visualization/__init__.py [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+if importlib.util.find_spec("weldx_widgets"):
+    from warnings import warn
+
+    from weldx_widgets.visualization.csm_k3d import (
+        CoordinateSystemManagerVisualizerK3D,
+        SpatialDataVisualizer,
+    )
+
+    warn("Using experimental 'weldx-widgets' implementation.", UserWarning)
+else:
```
Comment (marscher) [inline weldx/visualization/__init__.py:13]:
```suggestion
except ImportError:
```

File: weldx/visualization/__init__.py [hunk 2]
```diff
@@ -1,5 +1,18 @@
 """Visualization of spatial data, coordinate systems (cs), and cs manager."""
-from .k3d_impl import CoordinateSystemManagerVisualizerK3D, SpatialDataVisualizer
+import importlib.util
+
+if importlib.util.find_spec("weldx_widgets"):
```
Comment (marscher) [inline weldx/visualization/__init__.py:4]:
```suggestion
try:
```

**Comment by marscher** [comment]
I think we should also start to remove the matplotlib and k3d dependencies and add import-guards in the respective visualization modules. These guards should show an error message indicating that these dependencies have moved to weldx-widgets, or can be installed manually.

**Comment by CagtayFabry** [comment]
> I think we should also start to remove the matplotlib and k3d dependencies and add import-guards in the respective visualization modules. These guards should show an error message indicating that these dependencies have moved to weldx-widgets, or can be installed manually.

eventually yes, let's do this a little slower with the above approach but keep the default weldx behavior (and dependencies) stable for now

**Comment by marscher** [comment]
The approach I suggested would result in a pre-tested release with the conditions we would like to achieve. Otherwise I wonder how the road map for this change would look like? At some point we will do it any ways, right?

**Review by marscher — APPROVED** [review]

**Pull request merged**